src/selectors/per-thread/stack-sample.js

```diff
@@ -204,13 +215,33 @@ export function getStackAndSampleSelectorsPerThread(
     CallTree.computeCallTreeCountsAndTimings
   );
 
+  const _getCallTreeUnit = createSelector(
+    UrlState.getCallTreeSummaryStrategy,
+    strategy => {
+      switch (strategy) {
+        case 'timing':
+          return 'ms';
+        case 'js-allocations':
+        case 'native-allocations':
+        case 'native-deallocations':
+          return 'bytes';
+        default:
+          throw assertExhaustiveCheck(
+            strategy,
+            'Unhandled callTreeSummaryStrategy.'
+          );
+      }
+    }
+  );
```

Comment:
optional nit: I'm not convinced this should be in a selector.
And there's no need for the memoization.

So I'd rather move this to a more common function, that is more reusable and maybe easier to interface with a localization library.

If you really want to keep it a selector, maybe just create a normal function without `createSelector` that seems overkill here. Remember that memoization itself has a cost, and it's our job to evaluate whether memoization's cost is smaller than the function's cost.

Also see below: depending on where the values are used, we may need to change more than just the unit. That's why I think this switch should happen closer to where the strings are actually rendered, so that we have more context to better adapt the text.
(that's what you did for the call tree headers for example)

Reply:
That said, we could imagine that the string isn't the actual string to be rendered, but rather defines what "type" the data is...